infrastructure/src/main/java/org/corfudb/infrastructure/AbstractServer.java

```diff
@@ -34,7 +37,8 @@ public AbstractServer() {
      */
     public void handleMessage(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
         if (isShutdown()) return;
-        if (!getHandler().handle(msg, ctx, r)) {
+        boolean isMetricsEnabled = MetricsUtils.isMetricsCollectionEnabled();
+        if (!getHandler().handle(msg, ctx, r, isMetricsEnabled)) {
```

Comment:
you are passing a new flag to all handlers (isMetricEnabled), which I don't see used even once.

I see one place -- SequencerServer -- which needs it. As it turns out, SequencerServer is grabbing this directly from MetricsUtils.

The question is: Should we pass this parameter to all handlers, so that we can avoid that single usage, or the other way around? My sense is that perhaps we shouldn't pass it.